File: tests/benchmarks/resolution/fixtures/php/Repository.php
<?php

class User
{
    public string $id;
    public string $name;
    public string $email;

    public function __construct(string $id, string $name, string $email)
    {
        $this->id = $id;
        $this->name = $name;
        $this->email = $email;
    }
}

class UserRepository
{
    private array $store = [];

    public function findById(string $id): ?User
    {
        return $this->store[$id] ?? null;
    }

    public function save(User $user): void
    {
        $this->store[$user->id] = $user;
    }

    public function delete(string $id): bool
    {
        if (isset($this->store[$id])) {
            unset($this->store[$id]);
            return true;
        }
        return false;
    }
}

File: tests/benchmarks/resolution/fixtures/elixir/repository.ex
defmodule UserRepository do
  def new_store do
    %{}
  end

  def save(store, id, user) do
    Map.put(store, id, user)
  end

  def find_by_id(store, id) do
    Map.get(store, id)
  end

  def delete(store, id) do
    Map.delete(store, id)
  end

  def list_all(store) do
    Map.values(store)
  end
end
